doc/api/esm.md

```diff
@@ -1232,13 +1236,19 @@ The resolver has the following properties:
 * No default extensions
 * No folder mains
 * Bare specifier package resolution lookup through node\_modules
+* Does not fail on unknown extensions or protocols
 
 ### Resolver algorithm
 
 The algorithm to load an ES module specifier is given through the
 **ESM\_RESOLVE** method below. It returns the resolved URL for a
 module specifier relative to a parentURL.
 
+The algorithm does not determine whether the resolved URL protocol can be
+loaded by Node.js, because other loaders may be able to load it.
+The default loader _does_ fail if it was asked to load a URL
+that has an unsuppported protocol.
```

Comment:
Can we be clearer here about this point, since we are removing the error:

```
The Node.js default loader will throw an _Unsupported File Extension_ error if loading a file:/// URL that does not end in one of the supported file extensions, ".mjs", '.cjs", ".json"
```

Then it would help to add a note about the scheme handling here as well:

```
The default loader will also throw an _Unsupported_URL_Scheme_ error if loading a URL that is not of a file:///, data or node: builtin scheme.

Custom loaders are able to bypass these errors.
```

Or something along those lines.

Reply:
I modified the text to specify directly what protocols the default loader supports.

I added a similar paragraph that deals with the supported extensions, and which roughly says the same thing: resolver does not fail on unsupported extensions, loader does.